db/view/view.cc

```diff
@@ -2549,7 +2549,11 @@ void view_builder::on_drop_view(const sstring& ks_name, const sstring& view_name
 }
 
 future<> view_builder::do_build_step() {
-    return seastar::async([this] {
+    // Run the view building in the streaming scheduling group
+    // so that it doesn't impact other tasks with higher priority.
+    seastar::thread_attributes attr;
+    attr.sched_group = _db.get_streaming_scheduling_group();
+    return seastar::async(std::move(attr), [this] {
```

Comment:
I think it's better to move this to start_in_background() so the outer loop is included.

Are you sure it's not a regression? I think it's a regression from the point where the maintenance group and the gossip group were split.

Reply:
> I think it's better to move this to start_in_background() so the outer loop is included.

We can't just move this to `start_in_background` because `do_build_step` is called from both `start_in_background` and `view_builder::on_create_view`. I can put this in both places but that would move just a couple of extra operations to the streaming group, not scaling with the table size

> Are you sure it's not a regression? I think it's a regression from the point where the maintenance group and the gossip group were split.

From what I'm seeing, before raft, view building was triggered on a schema modification statement, so probably in the user scheduling group, which may be less bad than building in gossip group. It's still quite old already and doesn't cause test fails (that we know of), so I'm not sure it mandates a backport

Reply:
Perhaps we should consider changing the `serialized_action`'s interface to accept a scheduling group in the constructor and that scheduling group would be used to run the action.

I don't think we should do this in this PR, though.

Reply:
> Perhaps we should consider changing the serialized_action's interface to accept a scheduling group in the constructor and that scheduling group would be used to run the action.

I don't think we should mix orthogonal concepts.

> It's still quite old already and doesn't cause test fails (that we know of), so I'm not sure it mandates a backport

Backport consideration is not just test failures, but impact on production (if it's a regression).